fiaas_deploy_daemon/deployer/kubernetes/role_binding.py

```diff
@@ -0,0 +1,100 @@
+        self._clean_not_needed_role_bindings(role_bindings)
+        counter = 1
+        self._create_role_bindings(app_spec, self._list_of_roles, "Role", counter, custom_annotations, custom_labels, role_bindings)
+        self._create_role_bindings(app_spec, self._list_of_cluster_roles, "ClusterRole", counter, custom_annotations, custom_labels,
+                                   role_bindings)
+
+    def _create_role_bindings(self, app_spec: AppSpec, roles_list, role_kind, counter, custom_annotations, custom_labels,
+                              role_bindings):
+        LOG.info("Creating RoleBindings for %s", app_spec.name)
+        namespace = app_spec.namespace
+
+        service_account_name = app_spec.name
+
+        for role_name in roles_list:
+            role_binding = self._find_role_in_role_bindings(role_kind, role_name, role_bindings)
+            if role_binding:
+                if self._owned_by_fiaas(role_binding):
+                    role_binding.metadata = ObjectMeta(
+                        name=role_binding.metadata.name, namespace=namespace, labels=custom_labels, annotations=custom_annotations
+                    )
+                    role_ref = RoleRef(kind=role_kind, apiGroup="rbac.authorization.k8s.io", name=role_name)
+                    subject = Subject(kind="ServiceAccount", name=service_account_name, namespace=namespace)
+                    role_binding.roleRef = role_ref
+                    role_binding.subjects = [subject]
+                    self._owner_references.apply(role_binding, app_spec)
+                    role_binding.save()
+                else:
+                    LOG.info(
+                        "Aborting the creation of a roleBinding for Application: %s with %s: %s",
+                        app_spec.name,
+                        role_kind,
+                        role_name
+                    )
+            else:
+                while (True):
+                    role_binding_name = f"{app_spec.name}-{counter}"
+                    try:
+                        role_binding = RoleBinding.get(role_binding_name, namespace)
+                        counter += 1
+                    except NotFound:
+                        role_binding = RoleBinding()
+                        role_binding.metadata = ObjectMeta(
+                            name=role_binding_name, namespace=namespace, labels=custom_labels, annotations=custom_annotations
+                        )
+
+                        role_ref = RoleRef(kind=role_kind, apiGroup="rbac.authorization.k8s.io", name=role_name)
+                        subject = Subject(kind="ServiceAccount", name=service_account_name, namespace=namespace)
+                        role_binding.roleRef = role_ref
+                        role_binding.subjects = [subject]
+                        self._owner_references.apply(role_binding, app_spec)
+                        role_binding.save()
+                        counter += 1
```

Comment:
Although extracting to a method would help, I still think the flow would be much more clear if this part was moved outside of the loop. Should just be to move the break and change the indentation.

Reply:
Extracted to a separate method and moved it